crates/sparrow-session/src/session.rs

```diff
@@ -277,6 +277,58 @@ impl Session {
         }
     }
 
+    /// The [Expr] will call this to add a user-defined-function to the DFG directly.
+    ///
+    /// This bypasses much of the plumbing of the [ExprOp] required due to our construction
+    /// of the AST.
+    #[allow(unused)]
+    fn add_udf_to_dfg(
+        &mut self,
+        udf: Arc<dyn Udf>,
+        args: Vec<Expr>,
+    ) -> error_stack::Result<AstDfgRef, Error> {
+        let signature = udf.signature();
+        // TODO: Why do I have to do this?
+        // Why lifetime on arc now happy?
+        let arg_names = signature.arg_names().to_owned();
```

Comment:
Same problem on lifetimes for this

Reply:
see previous -- we can discuss further.

it should be possible to resolve, possibly by allowing the `Resolved` to take a lifetime? not sure if we want to fix now or just really tear into this later